libraries/mbed/targets/hal/TARGET_STM/TARGET_STM32F4/serial_api.c

```diff
@@ -141,18 +235,18 @@ void serial_init(serial_t *obj, PinName tx, PinName rx)
     }
 
     // Configure UART
-    obj->baudrate = 9600;
-    obj->databits = UART_WORDLENGTH_8B;
-    obj->stopbits = UART_STOPBITS_1;
-    obj->parity   = UART_PARITY_NONE;
+    _SERIAL_OBJ(baudrate) = 9600;
```

Comment:
I would not start a macro with underscore followed by an uppercase letter (it's reserved in any scope , see standard C/C++).

This macro is needed now as not all F4 are up to date to enable this feature ? Because this should not be needed if serial object is same for all F4 , it should be the same?